src/pool.rs

```diff
@@ -286,8 +342,7 @@ impl ConnectionPool {
                 Ok(conn) => conn,
                 Err(err) => {
                     error!("Banning instance {:?}, error: {:?}", address, err);
-                    self.ban(&address, address.shard, process_id);
-                    self.stats.client_disconnecting(process_id, address.id);
```

Comment:
I always wondered about this, we never really disconnected clients that failed to get connection from the pool. Good thing it is fixed now.